### ISSUE TITLE
CI: spark tests failing after pyarrow upgrade (0.15.0)

### ISSUE THREAD

**Issue by JulienPeloton** [astrolabsoftware/fink-broker#284]
**Describe the issue**
Recently, CI jobs started to fail with (log [here](https://travis-ci.org/astrolabsoftware/fink-broker/jobs/593913044))

```
java.lang.IllegalArgumentException
	at java.nio.ByteBuffer.allocate(ByteBuffer.java:334)
	at org.apache.arrow.vector.ipc.message.MessageSerializer.readMessage(MessageSerializer.java:543)
	at org.apache.arrow.vector.ipc.message.MessageChannelReader.readNext(MessageChannelReader.java:58)
...
```
and on the python side:

```
**********************************************************************
File "/home/travis/build/astrolabsoftware/fink-broker/fink_broker/filters.py", line 219, in __main__.apply_user_defined_filters
Failed example:
    df.select("decoded.candidate.*").show() # doctest: +NORMALIZE_WHITESPACE
Exception raised:
    Traceback (most recent call last):
      File "/opt/python/3.6.7/lib/python3.6/doctest.py", line 1330, in __run
        compileflags, 1), test.globs)
      File "<doctest __main__.apply_user_defined_filters[5]>", line 1, in <module>
        df.select("decoded.candidate.*").show() # doctest: +NORMALIZE_WHITESPACE
      File "/home/travis/build/astrolabsoftware/fink-broker/spark-2.4.1-bin-hadoop2.7/python/pyspark/sql/dataframe.py", line 378, in show
        print(self._jdf.showString(n, 20, vertical))
      File "/home/travis/build/astrolabsoftware/fink-broker/spark-2.4.1-bin-hadoop2.7/python/lib/py4j-0.10.7-src.zip/py4j/java_gateway.py", line 1257, in __call__
        answer, self.gateway_client, self.target_id, self.name)
      File "/home/travis/build/astrolabsoftware/fink-broker/spark-2.4.1-bin-hadoop2.7/python/pyspark/sql/utils.py", line 63, in deco
        return f(*a, **kw)
      File "/home/travis/build/astrolabsoftware/fink-broker/spark-2.4.1-bin-hadoop2.7/python/lib/py4j-0.10.7-src.zip/py4j/protocol.py", line 328, in get_return_value
        format(target_id, ".", name), value)
    py4j.protocol.Py4JJavaError: An error occurred while calling o151.showString.
```

Note that the first failure appears while no change to the python codebase was introduced (8f353a4190fbeb207689ddf2416825d35d11de4a). After looking carefully at the travis logs, I suspect this is due to a recent upgrade of pyarrow:

Last successful commit:
```
Successfully installed Cython-0.29.13 aiohttp-3.6.1 asn1crypto-1.0.0 astropy-3.2.1 astroquery-0.3.10 
async-timeout-3.0.1 avro-python3-1.9.1 beautifulsoup4-4.8.0 cffi-1.12.3 chardet-3.0.4 
codecov-2.0.15 confluent-kafka-1.2.0 coverage-4.5.4 coveralls-1.8.2 cryptography-2.7 
docopt-0.6.2 entrypoints-0.3 fastavro-0.22.5 html5lib-1.0.1 idna-2.8 idna-ssl-1.1.0 
jeepney-0.4.1 keyring-19.2.0 multidict-4.5.2 pandas-0.25.1 pyarrow-0.14.1 pycparser-2.19 
python-dateutil-2.8.0 pytz-2019.2 requests-2.22.0 secretstorage-3.1.1 slackclient-2.2.0 
soupsieve-1.9.4 typing-extensions-3.7.4 urllib3-1.25.6 webencodings-0.5.1 yarl-1.3.0
```

First failing commit:
```
Successfully installed Cython-0.29.13 aiohttp-3.6.1 asn1crypto-1.0.1 astropy-3.2.1 astroquery-0.3.10 
async-timeout-3.0.1 avro-python3-1.9.1 beautifulsoup4-4.8.1 cffi-1.12.3 chardet-3.0.4 
codecov-2.0.15 confluent-kafka-1.2.0 coverage-4.5.4 coveralls-1.8.2 cryptography-2.7 
docopt-0.6.2 entrypoints-0.3 fastavro-0.22.5 html5lib-1.0.1 idna-2.8 idna-ssl-1.1.0 
jeepney-0.4.1 keyring-19.2.0 multidict-4.5.2 pandas-0.25.1 pyarrow-0.15.0 pycparser-2.19 
python-dateutil-2.8.0 pytz-2019.2 requests-2.22.0 secretstorage-3.1.1 slackclient-2.2.0 
soupsieve-1.9.4 typing-extensions-3.7.4 urllib3-1.25.6 webencodings-0.5.1 yarl-1.3.0
```

the most relevant change is `pyarrow-0.14.1` -> `pyarrow-0.15.0`, that has been upgraded on Oct 05. 

Action item: check current code version with lower pyarrow version to see if the failure is still there.

**Comment by JulienPeloton** [comment]
The failure is indeed due to the recent pyarrow upgrade (see #285 )

**Comment by JulienPeloton** [comment]
Fixed in #285

**Comment by JulienPeloton** [comment]
Note that the problem indeed comes from the new pyarrow release a a temporary fix has been released: see https://issues.apache.org/jira/projects/SPARK/issues/SPARK-29367?filter=addedrecently

We will stick to pyarrow 0.14.1 however as long as a compatible version is not released.
